tests/ui/test_ui.py

```diff
@@ -68,18 +71,36 @@ def test_set_footer_text_specific_text(self, view, text='blah'):
 
     def test_set_footer_text_with_duration(self, view, mocker,
                                            custom_text="custom", duration=5.3):
-        mocker.patch('zulipterminal.ui.View.get_random_help',
-                     return_value=['some help text'])
-        mock_sleep = mocker.patch('time.sleep')
+        mocker.patch('zulipterminal.ui.View._reset_footer_text')
 
         view.set_footer_text([custom_text], duration)
 
-        view._w.footer.set_text.assert_has_calls([
-            mocker.call([custom_text]),
-            mocker.call(['some help text'])
-        ])
+        view._w.footer.set_text.assert_called_once_with([custom_text])
+        view.controller.update_screen.assert_called_once_with()
+        view._reset_footer_text.assert_called_once_with(duration)
+
+    def test__reset_footer_text(self, view, mocker, duration=46.2):
+        mock_sleep = mocker.patch('time.sleep')
+        mocker.patch('zulipterminal.ui.View.set_footer_text')
+
+        view._reset_footer_text(duration)
+
         mock_sleep.assert_called_once_with(duration)
-        assert view.controller.update_screen.call_count == 2
+        view.set_footer_text.assert_called_once_with()
+
+    def test__reset_footer_text_multiple_reset_attempts(self, view,
+                                                        mocker):
+        time.sleep = mocker.Mock(side_effect=time.sleep)
+        mocker.patch('zulipterminal.ui.View.set_footer_text')
+
+        # Multiple simultaneous calls for resetting footer.
+        with ThreadPoolExecutor(max_workers=3) as executor:
+            executor.submit(view._reset_footer_text, duration=0.5)
+            executor.submit(view._reset_footer_text, duration=0.5)
+            executor.submit(view._reset_footer_text, duration=0.5)
+
+        view.set_footer_text.assert_called_once_with()
+        assert time.sleep.call_count == 3
```

Comment:
I tried something different here - this is more of an integration test rather than the unit tests we have everywhere.

Reply:
(Also, pytest cannot call methods asynchronously?)